samza-core/src/main/java/org/apache/samza/clustermanager/ContainerProcessManager.java

```diff
@@ -409,32 +411,34 @@ public void onResourceCompleted(SamzaResourceStatus resourceStatus) {
             lastFailureTime = failure.getLastFailure();
           } else {
             currentFailCount = 1;
-            lastFailureTime = 0L;
+            lastFailureTime = Instant.now().toEpochMilli();
           }
-          if (currentFailCount >= retryCount) {
-            long lastFailureMsDiff = System.currentTimeMillis() - lastFailureTime;
-
-            if (lastFailureMsDiff < retryWindowMs) {
-              log.error("Processor ID: {} (current Container ID: {}) has failed {} times, with last failure {} ms ago. " +
-                  "This is greater than retry count of {} and window of {} ms, " +
-                  "so shutting down the application master and marking the job as failed.",
-                  processorId, containerId, currentFailCount, lastFailureMsDiff, retryCount, retryWindowMs);
-
-              // We have too many failures, and we're within the window
-              // boundary, so reset shut down the app master.
-              tooManyFailedContainers = true;
-              state.status = SamzaApplicationState.SamzaAppStatus.FAILED;
-            } else {
-              log.info("Resetting failure count for Processor ID: {} back to 1, since last failure " +
-                  "(for Container ID: {}) was outside the bounds of the retry window.", processorId, containerId);
-
-              // Reset counter back to 1, since the last failure for this
-              // container happened outside the window boundary.
-              processorFailures.put(processorId, new ProcessorFailure(1, System.currentTimeMillis()));
-            }
+
+          long lastFailureMsDiff = Instant.now().toEpochMilli() - lastFailureTime;
```

Comment:
Minor: Might be preferable to set lastFailureMsDiff in the if-condition above, to avoid time-skew in measurements.

Reply:
I imagine it would be a sub-millisecond skew? Would that matter? Will change anyways to make the code cleaner.

Reply:
Typically, yes. In some cases, CPU preemption can cause the time to be relatively large, i've only seen it once when the host was so busy it was spending most of the time context-switching